client/scripts/views/pages/create_community/cosmos_form.tsx

```diff
@@ -0,0 +1,120 @@
+/* @jsx m */
+
+import m from 'mithril';
+import $ from 'jquery';
+
+import 'pages/create_community.scss';
+
+import app from 'state';
+import { initAppState } from 'app';
+import { slugifyPreserveDashes } from 'utils';
+import { ChainBase, ChainType } from 'types';
+import { initChainForm, defaultChainRows } from './chain_input_rows';
+import { ChainFormFields, ChainFormState, EthFormFields } from './types';
+import { IdRow, InputRow, ValidationRow } from '../../components/metadata_rows';
+import { CWButton } from '../../components/component_kit/cw_button';
+
+// TODO: populate additional fields
+
+type CosmosFormFields = {
+  bech32_prefix: string;
+  decimals: number;
+};
+
+type CreateCosmosForm = ChainFormFields & EthFormFields & CosmosFormFields;
+
+type CreateCosmosState = ChainFormState & { form: CreateCosmosForm };
+
+export class CosmosForm implements m.ClassComponent {
+  private state: CreateCosmosState = {
+    error: '',
+    saving: false,
+    form: {
+      alt_wallet_url: '',
+      bech32_prefix: '',
+      decimals: 6,
+      id: '',
+      name: '',
+      symbol: 'XYZ',
+      node_url: '',
```

Comment:
🐫

Reply:
hmm i can also see how this is helpful in the `...this.state.form` in the post request... @gdjohnson @jnaviask, we want to have camelCase in the frontend, but this seems like a clean use of it for the request. but also, if we want a clean standard, then we should probably just have the post request `node_url: this.state.form.nodeUrl` or have it destructured earlier.

Reply:
in the sputnik page, the logic uses camelCases, destructures and then reassigns to the correct snake_case for the route. I think we should do that here too.